main.js

```diff
@@ -1,64 +1,81 @@
-// Define the book collection array
-let books = [];
+class BookCollection {
+  constructor() {
+    // Initialize an empty array for the books
+    this.books = [];
 
-// Get the input fields and buttons
-const titleInput = document.getElementById('title');
-const authorInput = document.getElementById('author');
-const addButton = document.getElementById('add');
-const removeButton = document.getElementById('remove');
+    // Get references to the input fields and buttons
+    this.titleInput = document.getElementById('title');
+    this.authorInput = document.getElementById('author');
+    this.addButton = document.getElementById('add');
+    this.removeButton = document.getElementById('remove');
 
-// Get the books list and render it on the page
-const booksList = document.querySelector('#books ul');
+    // Get a reference to the books list and render it on the page
+    this.booksList = document.querySelector('#books ul');
+    this.loadBooks();
+    this.addEventListeners();
+  }
 
-function renderBooks() {
-  booksList.innerHTML = '';
-  books.forEach((book, index) => {
-    const listItem = document.createElement('li');
-    listItem.innerHTML = `${book.title} by ${book.author}<button data-index="${index}">Remove</button>`;
-    booksList.appendChild(listItem);
-  });
-}
+  // Load the saved books from localStorage
+  loadBooks() {
+    if (localStorage.getItem('books')) {
+      this.books = JSON.parse(localStorage.getItem('books'));
+      this.renderBooks();
+    }
+  }
 
-// Add a book to the collection
-function addBook(title, author) {
-  const book = { title, author };
-  books.push(book);
-  renderBooks();
-}
+  // Get the books list and render it on the page
+  renderBooks() {
+    this.booksList.innerHTML = '';
+    this.books.forEach((book, index) => {
+      const listItem = document.createElement('li');
+      listItem.innerHTML = `"${book.title}" by ${book.author}<button data-index="${index}">Remove</button>`;
+      this.booksList.appendChild(listItem);
+    });
+  }
 
-// Remove a book from the collection
-function removeBook(index) {
-  books = books.filter((book, i) => i !== index);
-  renderBooks();
-}
+  // Add a book to the collection
+  addBook(title, author) {
+    const book = { title, author };
+    this.books.push(book);
```

Comment:
# Optional:
- You have done a nice job, congratulations. I would advise you to use the method that doesn't mutate the array, e.g. use [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push).
- [See code example](https://www.educative.io/answers/spread-operator-vs-arrayprototypeconcat-in-javascript)